src/ynEIGEN/ynEigen.sol

```diff
@@ -0,0 +1,320 @@
+        address admin;
+        address pauser;
+        address unpauser;
+        address[] pauseWhitelist;
+    }
+
+    function initialize(Init calldata init)
+        public
+        notZeroAddress(address(init.assetRegistry))
+        notZeroAddress(address(init.admin))
+        initializer {
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        __ynBase_init(init.name, init.symbol);
+
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(PAUSER_ROLE, init.pauser);
+        _grantRole(UNPAUSER_ROLE, init.unpauser);
+
+        assetRegistry = init.assetRegistry;
+        eigenStrategyManager = init.eigenStrategyManager;
+
+        _setTransfersPaused(true);  // transfers are initially paused
+        _updatePauseWhitelist(init.pauseWhitelist, true);
+    }
+
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  DEPOSITS   ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Deposits a specified amount of an asset into the contract and mints shares to the receiver.
+     * @dev This function first checks if the asset is supported, then converts the asset amount to unitOfAccount equivalent,
+     * calculates the shares to be minted based on the unitOfAccount value, mints the shares to the receiver, and finally
+     * transfers the asset from the sender to the contract. Emits a Deposit event upon success.
+     * @param asset The ERC20 asset to be deposited.
+     * @param amount The amount of the asset to be deposited.
+     * @param receiver The address to receive the minted shares.
+     * @return shares The amount of shares minted to the receiver.
+     */
+    function deposit(
+        IERC20 asset,
+        uint256 amount,
+        address receiver
+    ) public nonReentrant returns (uint256 shares) {
+
+        if (depositsPaused) {
+            revert Paused();
+        }
+
+        return _deposit(asset, amount, receiver, msg.sender);
+    }
+
+    function _deposit(
+        IERC20 asset,
+        uint256 amount,
+        address receiver,
+        address sender
+    ) internal returns (uint256 shares) {
+
+        if (!assetIsSupported(asset)) {
+            revert UnsupportedAsset(asset);
+        }
+
+        if (amount == 0) {
+            revert ZeroAmount();
+        }
+
+        asset.safeTransferFrom(sender, address(this), amount);
+
+        // Convert the value of the asset deposited to unitOfAccount
+        uint256 assetAmountInUnitOfAccount = assetRegistry.convertToUnitOfAccount(asset, amount);
+        // Calculate how many shares to be minted using the same formula as ynUnitOfAccount
+        shares = _convertToShares(assetAmountInUnitOfAccount, Math.Rounding.Floor);
+
+        // Mint the calculated shares to the receiver 
+        _mint(receiver, shares);
```

Comment:
This mint function doesn't revert if zero shares are minted, a minimal deposit, or in case of an attack resulting in zero shares being minted will result in a loss from the depositor.

Reply:
Addressed with check for zero, good point